src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -348,7 +358,14 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
 
         aMessage.Read(offset, sizeof(Tlv), &data.tlv);
         aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
-        mLocal.Set(data.tlv);
+
+        if (data.tlv.GetType() != Tlv::kChannel && data.tlv.GetType() != Tlv::kPanId &&
+            data.tlv.GetType() != Tlv::kNetworkMasterKey &&
+            data.tlv.GetType() != Tlv::kMeshLocalPrefix)
+        {
+            mLocal.Set(data.tlv);
```

Comment:
ok, i see your point. Previous what i understand about "changes" is that one or more Tlvs that would affect connectivity in MGMT_ACTIVE_SET request message change. Here "change" represents MGMT_ACTIVE_SET request message itself. That makes sense. Thanks for correcting me. I will update this soon.